src/prefect/agent/docker/agent.py

```diff
@@ -325,24 +326,41 @@ def deploy_flow(self, flow_run: GraphQLResult) -> str:
         # the 'import prefect' time low
         import docker
 
-        storage = StorageSchema().load(flow_run.flow.storage)
-        if not isinstance(StorageSchema().load(flow_run.flow.storage), Docker):
-            self.logger.error(
-                "Storage for flow run {} is not of type Docker.".format(flow_run.id)
-            )
-            raise ValueError("Unsupported Storage type")
+        image = None
+        registry_url = False
+
+        # First check if `image` found in environment metadata, then default to storage
+        environment = EnvironmentSchema().load(flow_run.flow.environment)
+        if hasattr(environment, "metadata") and hasattr(environment.metadata, "image"):
```

Comment:
Curious about this - `image` seems like such a fundamental part of the environment, are we sure we want to keep it one level removed in this `metadata` field?

Reply:
I don't have a big opinion on it but I would say that having things like this in a `metadata` field is nice from a serialization perspective since it wouldn't require new fields if it were to change in the future (all inside a json block). cc @jcrist 

Not to say that there couldn't be two new fields though `image` and `metadata`. Regardless, I'm a fan of a metadata field for generic information store.

Reply:
Interesting - that's a good point, the forwards / backwards compatibility provided by a JSON  blob is convenient